v2/apiserver/schemas/common.json

```diff
@@ -44,10 +44,6 @@
 				"ref": {
 					"type": "string",
 					"description": "A reference to a git branch or tag"
-				},
-				"initSubmodules": {
-					"type": "boolean",
-					"description": "Whether to initialize git submodules"
 				}
 			}
 		},
```

Comment:
This whole type isn't common anymore, should this def move closer to where it's used in event.json?

Reply:
The whole gitConfig type defined here _is_ common (event references it and only it, project references all of its properties and adds another)... I couldn't quite figure out how to consume the entire object in the latter case (project) via `additionalProperties: {...}` so I went the route of iterating through the common type's properties... lemme keep looking a bit as to whether it can be made more elegant.

Reply:
> project references all of its properties and adds another

I think I overlooked that.

Reply:
> lemme keep looking a bit as to whether it can be made more elegant.

Now that I'm reading it correctly, I happy as is.

But if you find something more elegant, that's good too.

Reply:
Updated to use `allOf` in tandem w/ `additionalProperties`.